### ISSUE TITLE
Deprecate visiting test

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2897]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#2897|
|Type|enhancement|
|Created|2007-11-13 22:08:04Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-visitor-2897|

We want to replace uses of this with decorators of various sorts.

<details><summary>Searchable metadata</summary>

```
trac-id__2897 2897
type__enhancement enhancement
reporter__jml jml
priority__highest highest
milestone__ 
branch__branches_deprecate_visitor_2897 branches/deprecate-visitor-2897
branch_author__ 
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1194991684000000 1194991684000000
changetime__1194992941000000 1194992941000000
version__None None
owner__jml jml

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

(In [21748]) Branching to 'deprecate-visitor-2897'

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)|@jml set owner to @radix|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix commented|
|-|-|

+1

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set owner to @jml|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_5"></a>|@jml set status to `closed`|
|-|-|

(In [21750]) Deprecate visit() on Test objects.

 * Fixes [#2897](https://github.com/twisted/twisted/issues/2897)
 * Author: jml
 * Reviewer: radix

Introduces warnings into the test suite which will be remedied shortly.
